src/transformers/models/seggpt/modeling_seggpt.py

```diff
@@ -0,0 +1,732 @@
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=std)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+        elif isinstance(module, SegGPTEncoder):
+            factor = self.config.initializer_range
+            nn.init.normal_(module.segment_token_x, mean=0.0, std=0.5 * factor)
+            nn.init.normal_(module.segment_token_y, mean=0.0, std=0.5 * factor)
+            nn.init.normal_(module.type_token_cls, mean=0.0, std=0.5 * factor)
+            nn.init.normal_(module.type_token_ins, mean=0.0, std=0.5 * factor)
+            nn.init.normal_(module.pos_embed, mean=0.0, std=0.5 * factor)
+
+
+@dataclass
+class SegGPTModelOutput(ModelOutput):
+    """
+    Class for outputs of [`MaskFormerForInstanceSegmentation`].
+
+    This output can be directly passed to [`~MaskFormerImageProcessor.post_process_semantic_segmentation`] or or
+    [`~MaskFormerImageProcessor.post_process_instance_segmentation`] or
+    [`~MaskFormerImageProcessor.post_process_panoptic_segmentation`] depending on the task. Please, see
+    [`~MaskFormerImageProcessor] for details regarding usage.
+
+    Args:
+        logits (`torch.FloatTensor` of shape `(batch_size, sequence_length, config.vocab_size)`):
+            Prediction of the decoder.
+        last_hidden_state (`torch.FloatTensor`):
+            Last hidden states (final feature map) of the last stage of the encoder model .
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings + one for the output of each stage) of
+            shape `(batch_size, num_channels, height, width)`. Hidden-states (also called feature maps) of the encoder
+            model at the output of each stage.
+        attentions (`tuple(torch.FloatTensor)`, *optional*, returned when `output_attentions=True` is passed or when `config.output_attentions=True`):
+            Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size, num_heads, sequence_length,
+            sequence_length)`. Attentions weights from Detr's decoder after the attention softmax, used to compute the
+            weighted average in the self-attention heads.
+    """
+
+    logits: torch.FloatTensor = None
+    last_hidden_state: Optional[torch.FloatTensor] = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+class SegGPTModel(SegGPTPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        # --------------------------------------------------------------------------
+        self.encoder = SegGPTEncoder(config)
+
+        self.decoder = SegGPTDecoder(config)
```

Comment:
The decoder needs to be part of the head model. `SegGPTModel` should only output `last_hidden_state`, `hidden_states` and `attentions` (similar to `BertModel`, `Gpt2Model`, etc.).

Reply:
Done.